pkg/registry/discovery/endpointslice/strategy.go

```diff
@@ -139,3 +145,37 @@ func dropDisabledFieldsOnUpdate(oldEPS, newEPS *discovery.EndpointSlice) {
 		}
 	}
 }
+
+// dropTopologyOnV1 on V1 request wipes the DeprecatedTopology field  and copies
+// the NodeName value into DeprecatedTopology
+func dropTopologyOnV1(ctx context.Context, oldEPS, newEPS *discovery.EndpointSlice) {
+	if info, ok := genericapirequest.RequestInfoFrom(ctx); ok {
+
+		// endpointListUpdated should be true for creates
+		endpointListUpdated := true
+		if oldEPS != nil {
+			endpointListUpdated = !apiequality.Semantic.DeepEqual(oldEPS.Endpoints, newEPS.Endpoints)
+		}
```

Comment:
nit: move this inside the version check so we don't do work we won't use on v1beta1 requests